sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -1608,4 +1608,40 @@ struct _pi_sampler : _pi_object {
   ze_sampler_handle_t ZeSampler;
 };
 
+struct _pi_ext_command_buffer : _pi_object {
+  _pi_ext_command_buffer(pi_context Context,
+                         ze_command_list_handle_t CommandList,
+                         ZeStruct<ze_command_list_desc_t> ZeDesc,
+                         const pi_ext_command_buffer_desc *Desc);
+
+  ~_pi_ext_command_buffer();
+
+  void RegisterSyncPoint(pi_ext_sync_point SyncPoint, pi_event Event) {
+    SyncPoints[SyncPoint] = Event;
+    NextSyncPoint++;
+  }
+
+  pi_ext_sync_point GetNextSyncPoint() const { return NextSyncPoint; }
+
+  // PI context associated with this command-buffer
+  pi_context Context;
+  // Level Zero command list handle
+  ze_command_list_handle_t ZeCommandList;
+  // Level Zero command list descriptor
+  ZeStruct<ze_command_list_desc_t> ZeCommandListDesc;
+  // Command buffer descriptor (TODO: Is it safe to store this? It contains
+  // pointers to host objects potentially)
+  pi_ext_command_buffer_desc CommandBufferDesc;
+  // Map of sync_points to pi_events
+  std::unordered_map<pi_ext_sync_point, pi_event> SyncPoints;
+  // Next sync_point value (may need to consider ways to reuse values if 32-bits
```

Comment:
Ah that's fair! I think I was considering whether we would want to make sync points unique across all command-buffers to prevent using a sync point from the wrong command-buffer, but that is probably not worth worrying about.